api/queue.py

```diff
@@ -0,0 +1,73 @@
+from django.db import connection
+
+from api.models import TaskDef, Task
+
+get_task_sql = """
```

Comment:
I could load some postgres specific django DSL, but I find them harder to read than SQL for a complex query like this. We're using a SQL backend, anyone programming in the backend should understand SQL.